### PR TITLE
Attaches to s3 in crew

### LINKED ISSUE #777: Attaches to s3 in crew

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #777 

## PR includes

-   [x] Feature

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -557,3 +557,10 @@ model GroupAdmin {
 
   @@unique([userId, groupId])
 }
+model Attach {
```
Comment (tsumo) [inline prisma/schema.prisma:560]: empty line between models

File: src/hooks/useRouter.ts [hunk 2]
```diff
@@ -20,6 +20,8 @@ export const pages = {
     scheduledDeactivations: '/deactivations',
 
     adminPanel: '/admin-panel',
+    attaches: '/api/attach',
+    attach: (attachId: string) => `/api/attach/${attachId}`,
```
Comment (tsumo) [inline src/hooks/useRouter.ts:24]: update useRouter below, please

File: src/modules/attachHandler.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+export const getHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+    const fileId = req.query.id;
+
+    const attach = await attachMethods.getById(String(fileId));
+
+    const file = await getObject(attach.link);
+
+    if (!file) throw new ErrorWithStatus(tr('No file finded'), 404);
```
Comment (tsumo) [inline src/modules/attachHandler.ts:81]: found

File: src/modules/attachMethods.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+};
+
+const deleteAttach = async (id: string): Promise<Attach> => {
+    await getById(id);
+
+    return prisma.attach.update({ where: { id }, data: { deletedAt: new Date() } });
+};
+
+export const attachMethods = {
+    create,
```
Comment (tsumo) [inline src/modules/attachMethods.ts:32]: lets move body of functions inside the object itself, this way it's one less click when you got from usage to definition in IDE

File: src/modules/attachModule.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { Attach } from '@prisma/client';
+
+import { pages } from '../hooks/useRouter';
+import { httpClient } from '../utils/httpClient';
+
+const remove = (id: string) => httpClient.delete<Attach>(pages.attach(id));
+
+export const attachModule = { remove };
```
Comment (tsumo) [inline src/modules/attachModule.ts:8]: why do we need this module?

File: src/pages/api/attach.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+export const config = {
+    api: {
+        bodyParser: false,
+    },
+};
+
+const handler = getApiHandler()
+    .use(getAuthChecker())
+    .delete(accessGuards.attach.delete, async (req, res) => {
+        const fileId = String(req.query.id);
```
Comment (tsumo) [inline src/pages/api/attach.ts:18]: Let's add history records for post and delete actions

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/utils/notifications/notificationsMap.ts [hunk 7]
```diff
@@ -172,6 +172,11 @@ export const getNotificicationKeyMap = (key: keyof NotificationMap) => {
             loading: tr('Canceling profile deactivation...'),
             error: tr('An error occurred while canceling profile deactivation'),
         },
+        attachDelete: {
```
Comment (tsumo) [inline src/utils/notifications/notificationsMap.ts:175]: I think this is `attachCreate`

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/utils/notifications/notificationsMap.ts (hunk 7 above)
Comment (nyet-ty) [reply src/utils/notifications/notificationsMap.ts:175]: it is, thanks

File: src/modules/attachAccess.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+export const accessGuards = {
+    attach: {
+        create: createGuard((session) => checkRoleForAccess(session.user.role, 'editScheduledDeactivation')),
+        readOne: createGuard(
+            (session) => checkRoleForAccess(session.user.role, 'editScheduledDeactivation'),
+            (session) => checkRoleForAccess(session.user.role, 'viewScheduledDeactivation'),
+        ),
+        delete: createGuard((session) => checkRoleForAccess(session.user.role, 'editScheduledDeactivation')),
+    },
```
Comment (asabotovich) [inline src/modules/attachAccess.ts:46]: check editScheduledDeactivation and viewScheduledDeactivation flags in place

File: src/pages/api/attach.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        const attach = await attachMethods.getById(String(fileId));
+
+        await removeFile(attach.link);
+        const result = await attachMethods.deleteAttach(fileId);
+        res.send(result);
+    })
```
Comment (asabotovich) [inline src/pages/api/attach.ts:24]: keep handlers in one place?

**Review by asabotovich — APPROVED** [review]

**Review by tsumo — APPROVED** [review]

**Pull request merged**